### ISSUE TITLE
issue when using restriction with 10 outcomes

### ISSUE THREAD

**Issue by SamSalvaggio** [bozenne/BuyseTest#10]
Hi Brice, 

For BuyseTest version 2.4.0.

I have an issue when I want to use a "restriction time" to more than 1 outcome.

The following code works :

```
BuyseTest(data = df,
                          arm ~ strat_column +
                            tte(tte1, status = cnsr1, threshold = 10, restriction = 365) +
                            tte(tte2, status = cnsr2, threshold = 10) +
                            tte(tte3, status = cnsr3, threshold = 10) +
                            tte(tte4, status = cnsr4, threshold = 10) +
                            bin(bin_var, operator = "<0") +
                            tte(tte5, status = cnsr5, threshold = 0) +
                            tte(tte6, status = cnsr6, threshold = 0) +
                            tte(tte7, status = cnsr7, threshold = 0) +
                            tte(tte8, status = cnsr8, threshold = 0) +
                            tte(tte9, status = cnsr9, threshold = 0),
                          trace = F, method.inference = "u-statistic", keep.pairScore = TRUE,
                          scoring.rule = "Peron", pool.strata = "CMH"
                          )
```

However, when adding the restriction to one or more outcomes such as this : 

```
BuyseTest(data = df,
                          arm ~ strat_column +
                            tte(tte1, status = cnsr1, threshold = 180, restriction = 365) +
                            tte(tte2, status = cnsr2, threshold = 180,  restriction = 365) +
                            tte(tte3, status = cnsr3, threshold = 180,  restriction = 365) +
                            tte(tte4, status = cnsr4, threshold = 180,  restriction = 365) +
                            bin(bin_var, operator = "<0") +
                            tte(tte5, status = cnsr5, threshold = 0,  restriction = 365) +
                            tte(tte6, status = cnsr6, threshold = 0,  restriction = 365) +
                            tte(tte7, status = cnsr7, threshold = 0,  restriction = 365) +
                            tte(tte8, status = cnsr8, threshold = 0,  restriction = 365) +
                            tte(tte9, status = cnsr9, threshold = 0,  restriction = 365),
                          trace = F, method.inference = "u-statistic", keep.pairScore = TRUE,
                          scoring.rule = "Peron", pool.strata = "CMH"
                          )

```
I obtain the following error :  

> Error in (function (name.call, status, correction.uninf, cpus, data, endpoint,  : 
>   Strata variable(s) "strat_column" "tte(tte9,status=cnsr9,
> threshold=0, restriction = 365)" not found in argument 'data'

which I don't understand as these columns are in "df". Sorry to bother you with this...

Kind regards,
Samuel

**Comment by bozenne** [comment]
Hello Samuel,

Humm, I tried to replicate the issue without success:

```
 set.seed(1)
    dt <- simBuyseTest(n.T = 50, n.C = 50,
                       names.strata = "strat_column", n.strata = 3,
                       argsTTE = list(name = c("tte1","tte2","tte3"), name.censoring = c("cnsr1","cnsr2","cnsr3"),
                                      scale.T = c(200,100,250), scale.censoring.T = c(10^5,10^5,10^5),
                                      scale.C = c(200,100,250), scale.censoring.C = c(10^5,10^5,10^5)),
                       argsBin = list(name = "bin_var"))

    formula1 <- treatment ~ strat_column + tte(tte1, status = cnsr1, threshold = 10, restriction = 365) + tte(tte2, status = cnsr2, threshold = 10, restriction = 365) + bin(bin_var, operator = "<0") + tte(tte3, status = cnsr3, threshold = 10, restriction = 365)

    GPC.v1 <- BuyseTest(formula1,
                        data = dt,
                        trace = FALSE)

    GPC.v2 <- BuyseTest(treatment = "treatment",
                        strata = "strat_column",
                        endpoint = c("tte1","tte2","bin_var","tte3"),
                        status = c("cnsr1","cnsr2","cnsr3"),
                        type = c("tte","tte","bin","tte"),
                        operator = c(">0",">0","<0",">0"),
                        threshold = c(10,10,NA,10),
                        restriction = c(365,365,NA,365),
                        data = dt,
                        trace = FALSE)
```

Could you provide a reproducible example so I can debug it?
Thanks!
Brice

**Comment by SamSalvaggio** [comment]
Thanks for your response.

Here is the code, it seems it works when specifying the formula similarly to "GPC.v2", however it returns the same error message when using the formula1 variable in GPC.v1

Code :

```
set.seed(1)
dt <- simBuyseTest(n.T = 50, n.C = 50,
                   names.strata = "strat_column", n.strata = 6,
                   argsTTE = list(name = c("tte1","tte2","tte3", "tte4", "tte5", "tte6", "tte7", "tte8", "tte9"), 
                                  name.censoring = c("cnsr1","cnsr2","cnsr3", "cnsr4", "cnsr5", "cnsr6", "cnsr7", "cnsr8", "cnsr9"),
                                  scale.T = c(200,100,250,300,150,200,350,400,450), scale.censoring.T = c(10^5,10^5,10^5,10^5,10^5,10^5,10^5,10^5,10^5),
                                  scale.C = c(200,100,250,300,150,200,350,400,450), scale.censoring.C = c(10^5,10^5,10^5,10^5,10^5,10^5,10^5,10^5,10^5)),
                   argsBin = list(name = "bin_var"))

formula1 <- treatment ~ strat_column + 
  tte(tte1, status = cnsr1, threshold = 10, restriction = 365) + 
  tte(tte2, status = cnsr2, threshold = 10, restriction = 365) + 
  tte(tte3, status = cnsr3, threshold = 10, restriction = 365) +
  tte(tte4, status = cnsr4, threshold = 10, restriction = 365) +
  bin(bin_var, operator = "<0") + 
  tte(tte5, status = cnsr5, threshold = 10, restriction = 365) + 
  tte(tte6, status = cnsr6, threshold = 10, restriction = 365) + 
  tte(tte7, status = cnsr7, threshold = 10, restriction = 365) +
  tte(tte8, status = cnsr8, threshold = 10, restriction = 365) +
  tte(tte9, status = cnsr9, threshold = 10, restriction = 365)

GPC.v1 <- BuyseTest(formula1,
                    data = dt,
                    trace = FALSE)

GPC.v2 <- BuyseTest(treatment = "treatment",
                    strata = "strat_column",
                    endpoint = c("tte1","tte2","tte3","tte4","bin_var","tte5","tte6","tte7","tte8","tte9"),
                    status = c("cnsr1","cnsr2","cnsr3","cnsr4","cnsr5","cnsr6","cnsr7","cnsr8","cnsr9"),
                    type = c("tte","tte","tte","tte","bin","tte","tte","tte","tte","tte"),
                    operator = c(">0",">0",">0",">0","<0",">0",">0",">0",">0",">0"),
                    threshold = c(10,10,10,10,NA,10,10,10,10,10),
                    restriction = c(365,365,365,365,NA,365,365,365,365,365),
                    data = dt,
                    trace = FALSE)


```

**Comment by bozenne** [comment]
Thanks! I can reproduce the error now. I'll have a look and will fix it - probably only next week though

**Comment by bozenne** [comment]
Fixed in the new version 2.4.3
